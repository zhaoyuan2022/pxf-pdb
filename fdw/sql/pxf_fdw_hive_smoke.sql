-- data prep
-- start_ignore
{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} mkdir -p {{ TEST_LOCATION }}
\!mkdir -p {{ TEST_LOCATION }}
COPY (
	SELECT format('row_%s',i::varchar(255)),
		i,
		i*0.0001,
		i*100000000000,
		CASE WHEN (i%2) = 0 THEN 'true' ELSE 'false' END
		from generate_series(1, 100) s(i)
	) TO '{{ TEST_LOCATION }}/data.csv'
	WITH (FORMAT 'csv');
{{ GPDB_REMOTE }}-- if GPDB is remote, will need to scp file down from there for beeline
{{ GPDB_REMOTE }}\!scp {{ PGHOST }}:{{ TEST_LOCATION }}/data.csv {{ TEST_LOCATION }}
{{ HIVE_REMOTE }}-- if hive is remote, will need to scp file up there to load it in
{{ HIVE_REMOTE }}\!cat {{ TEST_LOCATION }}/data.csv | ssh {{ HIVE_HOST }} 'mkdir -p {{ TEST_LOCATION }} && cat > {{ TEST_LOCATION }}/data.csv'
\!{{ BEELINE_CMD }} -f {{ SCRIPT create_hive_smoke_test_database.sql }} -u 'jdbc:hive2://{{ HIVE_HOST }}:10000/default{{ HIVE_PRINCIPAL }}'
\!{{ BEELINE_CMD }} -f {{ SCRIPT load_small_data.sql }} -u 'jdbc:hive2://{{ HIVE_HOST }}:10000/hive_smoke_test_database_{{ FULL_TESTNAME }}{{ HIVE_PRINCIPAL }}'
-- end_ignore

-- FDW test
CREATE SERVER hive_smoke_test_server
	FOREIGN DATA WRAPPER hive_pxf_fdw
	OPTIONS (config '{{ SERVER_CONFIG }}');
CREATE USER MAPPING FOR CURRENT_USER SERVER hive_smoke_test_server;
CREATE FOREIGN TABLE hive_smoke_test_foreign_table (
		name TEXT,
		num INTEGER,
		dub DOUBLE PRECISION,
		longNum BIGINT,
		bool BOOLEAN
	)
	SERVER hive_smoke_test_server
	OPTIONS (resource 'hive_smoke_test_database_{{ FULL_TESTNAME }}.hive_small_data_table_{{ FULL_TESTNAME }}');

-- @description query01 for PXF test on small data
SELECT * FROM hive_smoke_test_foreign_table ORDER BY name;

-- @description query02 for PXF test on small data
SELECT name, num FROM hive_smoke_test_foreign_table WHERE num > 50 ORDER BY name;

-- start_ignore
{{ CLEAN_UP }}-- clean up Hive and local disk
{{ CLEAN_UP }}\!rm -rf {{ TEST_LOCATION }}
{{ CLEAN_UP }}\!{{ BEELINE_CMD }} -f {{ SCRIPT cleanup_hive_smoke_test.sql }}
{{ CLEAN_UP }}\!rm -rf {{ SCRIPT cleanup_hive_smoke_test.sql }} {{ SCRIPT load_small_data.sql }} {{ SCRIPT create_hive_smoke_test_database.sql }}
{{ CLEAN_UP }}{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} rm -rf {{ TEST_LOCATION }}
-- end_ignore
