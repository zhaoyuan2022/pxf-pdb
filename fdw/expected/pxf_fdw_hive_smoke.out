-- data prep
-- start_ignore
{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} mkdir -p {{ TEST_LOCATION }}
\!mkdir -p {{ TEST_LOCATION }}
COPY (
	SELECT format('row_%s',i::varchar(255)),
		i,
		i*0.0001,
		i*100000000000,
		CASE WHEN (i%2) = 0 THEN 'true' ELSE 'false' END
		from generate_series(1, 100) s(i)
	) TO '{{ TEST_LOCATION }}/data.csv'
	WITH (FORMAT 'csv');
{{ GPDB_REMOTE }}-- if GPDB is remote, will need to scp file down from there for beeline
{{ GPDB_REMOTE }}\!scp {{ PGHOST }}:{{ TEST_LOCATION }}/data.csv {{ TEST_LOCATION }}
{{ HIVE_REMOTE }}-- if hive is remote, will need to scp file up there to load it in
{{ HIVE_REMOTE }}\!cat {{ TEST_LOCATION }}/data.csv | ssh {{ HIVE_HOST }} 'mkdir -p {{ TEST_LOCATION }} && cat > {{ TEST_LOCATION }}/data.csv'
\!{{ BEELINE_CMD }} -f {{ SCRIPT create_hive_smoke_test_database.sql }} -u 'jdbc:hive2://{{ HIVE_HOST }}:10000/default{{ HIVE_PRINCIPAL }}'
Connecting to jdbc:hive2://{{ HIVE_HOST }}:10000/default{{ HIVE_PRINCIPAL }}
Connected to: Apache Hive
Driver: Hive JDBC
Transaction isolation: TRANSACTION_REPEATABLE_READ
jdbc:hive2> CREATE DATABASE hive_smoke_test_database_{{ FULL_TESTNAME }};
jdbc:hive2> 
jdbc:hive2> Closing: 0: jdbc:hive2://{{ HIVE_HOST }}:10000/default{{ HIVE_PRINCIPAL }}
\!{{ BEELINE_CMD }} -f {{ SCRIPT load_small_data.sql }} -u 'jdbc:hive2://{{ HIVE_HOST }}:10000/hive_smoke_test_database_{{ FULL_TESTNAME }}{{ HIVE_PRINCIPAL }}'
Connecting to jdbc:hive2://{{ HIVE_HOST }}:10000/hive_smoke_test_database_{{ FULL_TESTNAME }}{{ HIVE_PRINCIPAL }}
Connected to: Apache Hive
Driver: Hive JDBC
Transaction isolation: TRANSACTION_REPEATABLE_READ
jdbc:hive2> CREATE TABLE hive_small_data_table_{{ FULL_TESTNAME }} (
jdbc:hive2> name string,
jdbc:hive2> num int,
jdbc:hive2> dub double,
jdbc:hive2> longNum bigint,
jdbc:hive2> bool boolean
jdbc:hive2> )
jdbc:hive2> ROW FORMAT DELIMITED FIELDS TERMINATED BY ','
jdbc:hive2> STORED AS textfile;
jdbc:hive2> LOAD DATA LOCAL INPATH '{{ TEST_LOCATION }}/data.csv'
jdbc:hive2> INTO TABLE  hive_small_data_table_{{ FULL_TESTNAME }};
jdbc:hive2> 
jdbc:hive2> Closing: 0: jdbc:hive2://{{ HIVE_HOST }}:10000/hive_smoke_test_database_{{ FULL_TESTNAME }}{{ HIVE_PRINCIPAL }}
-- end_ignore
-- FDW test
CREATE SERVER hive_smoke_test_server
	FOREIGN DATA WRAPPER hive_pxf_fdw
	OPTIONS (config '{{ SERVER_CONFIG }}');
CREATE USER MAPPING FOR CURRENT_USER SERVER hive_smoke_test_server;
CREATE FOREIGN TABLE hive_smoke_test_foreign_table (
		name TEXT,
		num INTEGER,
		dub DOUBLE PRECISION,
		longNum BIGINT,
		bool BOOLEAN
	)
	SERVER hive_smoke_test_server
	OPTIONS (resource 'hive_smoke_test_database_{{ FULL_TESTNAME }}.hive_small_data_table_{{ FULL_TESTNAME }}');
-- @description query01 for PXF test on small data
SELECT * FROM hive_smoke_test_foreign_table ORDER BY name;
  name   | num |  dub   |    longnum     | bool 
---------+-----+--------+----------------+------
 row_1   |   1 | 0.0001 |   100000000000 | f
 row_10  |  10 |  0.001 |  1000000000000 | t
 row_100 | 100 |   0.01 | 10000000000000 | t
 row_11  |  11 | 0.0011 |  1100000000000 | f
 row_12  |  12 | 0.0012 |  1200000000000 | t
 row_13  |  13 | 0.0013 |  1300000000000 | f
 row_14  |  14 | 0.0014 |  1400000000000 | t
 row_15  |  15 | 0.0015 |  1500000000000 | f
 row_16  |  16 | 0.0016 |  1600000000000 | t
 row_17  |  17 | 0.0017 |  1700000000000 | f
 row_18  |  18 | 0.0018 |  1800000000000 | t
 row_19  |  19 | 0.0019 |  1900000000000 | f
 row_2   |   2 | 0.0002 |   200000000000 | t
 row_20  |  20 |  0.002 |  2000000000000 | t
 row_21  |  21 | 0.0021 |  2100000000000 | f
 row_22  |  22 | 0.0022 |  2200000000000 | t
 row_23  |  23 | 0.0023 |  2300000000000 | f
 row_24  |  24 | 0.0024 |  2400000000000 | t
 row_25  |  25 | 0.0025 |  2500000000000 | f
 row_26  |  26 | 0.0026 |  2600000000000 | t
 row_27  |  27 | 0.0027 |  2700000000000 | f
 row_28  |  28 | 0.0028 |  2800000000000 | t
 row_29  |  29 | 0.0029 |  2900000000000 | f
 row_3   |   3 | 0.0003 |   300000000000 | f
 row_30  |  30 |  0.003 |  3000000000000 | t
 row_31  |  31 | 0.0031 |  3100000000000 | f
 row_32  |  32 | 0.0032 |  3200000000000 | t
 row_33  |  33 | 0.0033 |  3300000000000 | f
 row_34  |  34 | 0.0034 |  3400000000000 | t
 row_35  |  35 | 0.0035 |  3500000000000 | f
 row_36  |  36 | 0.0036 |  3600000000000 | t
 row_37  |  37 | 0.0037 |  3700000000000 | f
 row_38  |  38 | 0.0038 |  3800000000000 | t
 row_39  |  39 | 0.0039 |  3900000000000 | f
 row_4   |   4 | 0.0004 |   400000000000 | t
 row_40  |  40 |  0.004 |  4000000000000 | t
 row_41  |  41 | 0.0041 |  4100000000000 | f
 row_42  |  42 | 0.0042 |  4200000000000 | t
 row_43  |  43 | 0.0043 |  4300000000000 | f
 row_44  |  44 | 0.0044 |  4400000000000 | t
 row_45  |  45 | 0.0045 |  4500000000000 | f
 row_46  |  46 | 0.0046 |  4600000000000 | t
 row_47  |  47 | 0.0047 |  4700000000000 | f
 row_48  |  48 | 0.0048 |  4800000000000 | t
 row_49  |  49 | 0.0049 |  4900000000000 | f
 row_5   |   5 | 0.0005 |   500000000000 | f
 row_50  |  50 |  0.005 |  5000000000000 | t
 row_51  |  51 | 0.0051 |  5100000000000 | f
 row_52  |  52 | 0.0052 |  5200000000000 | t
 row_53  |  53 | 0.0053 |  5300000000000 | f
 row_54  |  54 | 0.0054 |  5400000000000 | t
 row_55  |  55 | 0.0055 |  5500000000000 | f
 row_56  |  56 | 0.0056 |  5600000000000 | t
 row_57  |  57 | 0.0057 |  5700000000000 | f
 row_58  |  58 | 0.0058 |  5800000000000 | t
 row_59  |  59 | 0.0059 |  5900000000000 | f
 row_6   |   6 | 0.0006 |   600000000000 | t
 row_60  |  60 |  0.006 |  6000000000000 | t
 row_61  |  61 | 0.0061 |  6100000000000 | f
 row_62  |  62 | 0.0062 |  6200000000000 | t
 row_63  |  63 | 0.0063 |  6300000000000 | f
 row_64  |  64 | 0.0064 |  6400000000000 | t
 row_65  |  65 | 0.0065 |  6500000000000 | f
 row_66  |  66 | 0.0066 |  6600000000000 | t
 row_67  |  67 | 0.0067 |  6700000000000 | f
 row_68  |  68 | 0.0068 |  6800000000000 | t
 row_69  |  69 | 0.0069 |  6900000000000 | f
 row_7   |   7 | 0.0007 |   700000000000 | f
 row_70  |  70 |  0.007 |  7000000000000 | t
 row_71  |  71 | 0.0071 |  7100000000000 | f
 row_72  |  72 | 0.0072 |  7200000000000 | t
 row_73  |  73 | 0.0073 |  7300000000000 | f
 row_74  |  74 | 0.0074 |  7400000000000 | t
 row_75  |  75 | 0.0075 |  7500000000000 | f
 row_76  |  76 | 0.0076 |  7600000000000 | t
 row_77  |  77 | 0.0077 |  7700000000000 | f
 row_78  |  78 | 0.0078 |  7800000000000 | t
 row_79  |  79 | 0.0079 |  7900000000000 | f
 row_8   |   8 | 0.0008 |   800000000000 | t
 row_80  |  80 |  0.008 |  8000000000000 | t
 row_81  |  81 | 0.0081 |  8100000000000 | f
 row_82  |  82 | 0.0082 |  8200000000000 | t
 row_83  |  83 | 0.0083 |  8300000000000 | f
 row_84  |  84 | 0.0084 |  8400000000000 | t
 row_85  |  85 | 0.0085 |  8500000000000 | f
 row_86  |  86 | 0.0086 |  8600000000000 | t
 row_87  |  87 | 0.0087 |  8700000000000 | f
 row_88  |  88 | 0.0088 |  8800000000000 | t
 row_89  |  89 | 0.0089 |  8900000000000 | f
 row_9   |   9 | 0.0009 |   900000000000 | f
 row_90  |  90 |  0.009 |  9000000000000 | t
 row_91  |  91 | 0.0091 |  9100000000000 | f
 row_92  |  92 | 0.0092 |  9200000000000 | t
 row_93  |  93 | 0.0093 |  9300000000000 | f
 row_94  |  94 | 0.0094 |  9400000000000 | t
 row_95  |  95 | 0.0095 |  9500000000000 | f
 row_96  |  96 | 0.0096 |  9600000000000 | t
 row_97  |  97 | 0.0097 |  9700000000000 | f
 row_98  |  98 | 0.0098 |  9800000000000 | t
 row_99  |  99 | 0.0099 |  9900000000000 | f
(100 rows)

-- @description query02 for PXF test on small data
SELECT name, num FROM hive_smoke_test_foreign_table WHERE num > 50 ORDER BY name;
  name   | num 
---------+-----
 row_100 | 100
 row_51  |  51
 row_52  |  52
 row_53  |  53
 row_54  |  54
 row_55  |  55
 row_56  |  56
 row_57  |  57
 row_58  |  58
 row_59  |  59
 row_60  |  60
 row_61  |  61
 row_62  |  62
 row_63  |  63
 row_64  |  64
 row_65  |  65
 row_66  |  66
 row_67  |  67
 row_68  |  68
 row_69  |  69
 row_70  |  70
 row_71  |  71
 row_72  |  72
 row_73  |  73
 row_74  |  74
 row_75  |  75
 row_76  |  76
 row_77  |  77
 row_78  |  78
 row_79  |  79
 row_80  |  80
 row_81  |  81
 row_82  |  82
 row_83  |  83
 row_84  |  84
 row_85  |  85
 row_86  |  86
 row_87  |  87
 row_88  |  88
 row_89  |  89
 row_90  |  90
 row_91  |  91
 row_92  |  92
 row_93  |  93
 row_94  |  94
 row_95  |  95
 row_96  |  96
 row_97  |  97
 row_98  |  98
 row_99  |  99
(50 rows)

-- start_ignore
{{ CLEAN_UP }}-- clean up Hive and local disk
{{ CLEAN_UP }}\!rm -rf {{ TEST_LOCATION }}
{{ CLEAN_UP }}\!{{ BEELINE_CMD }} -f {{ SCRIPT cleanup_hive_smoke_test.sql }}
{{ CLEAN_UP }}beeline> !connect 'jdbc:hive2://{{ HIVE_HOST }}:10000/default{{ HIVE_PRINCIPAL }}' "" "" ""
{{ CLEAN_UP }}Connecting to jdbc:hive2://{{ HIVE_HOST }}:10000/default{{ HIVE_PRINCIPAL }}
{{ CLEAN_UP }}Connected to: Apache Hive
{{ CLEAN_UP }}Driver: Hive JDBC
{{ CLEAN_UP }}Transaction isolation: TRANSACTION_REPEATABLE_READ
{{ CLEAN_UP }}jdbc:hive2> DROP DATABASE hive_smoke_test_database_{{ FULL_TESTNAME }} CASCADE;
{{ CLEAN_UP }}jdbc:hive2> 
{{ CLEAN_UP }}jdbc:hive2> Closing: 0: jdbc:hive2://{{ HIVE_HOST }}:10000/default{{ HIVE_PRINCIPAL }}
{{ CLEAN_UP }}\!rm -rf {{ SCRIPT cleanup_hive_smoke_test.sql }} {{ SCRIPT load_small_data.sql }} {{ SCRIPT create_hive_smoke_test_database.sql }}
{{ CLEAN_UP }}{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} rm -rf {{ TEST_LOCATION }}
-- end_ignore
